src/plugins/telemetry/server/plugin.ts

```diff
@@ -197,7 +198,14 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
       router,
       telemetryCollectionManager,
       savedObjectsInternalClient$: this.savedObjectsInternalClient$,
-      getSecurity: () => this.security,
+      getSecurity: () => {
+        // The security plugin is always enabled in X-Pack.
+        // However, if ES has `xpack.security.enabled: false`, some APIs might throw.
+        // We want only want to return the security plugin if we can confirm it's enabled.
+        if (security?.license.isEnabled()) {
+          return this.security;
+        }
+      },
```

Comment:
I noticed that when running Kibana with `xpack.security.enabled: false`, users couldn't fetch the telemetry sample because it failed with 
```JSON
{"error":"no handler found for uri [/_security/user/_has_privileges] and method [POST]"}
```

The reason: calling `security.authz.checkPrivilegesWithRequest()` when ES's security is disabled here: https://github.com/elastic/kibana/blob/a5b2db4604b45919a9b1e82e93b7b921c7d7ad56/src/plugins/telemetry/server/routes/telemetry_usage_stats.ts#L48-L60

Reply:
Please, note that this didn't affect collecting telemetry because in that case it's encrypted and skips the security validation.